src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -262,26 +274,47 @@ public ResponseEntity<byte[]> getCourseIcon(@PathVariable Long courseId, @PathVa
     }
 
     /**
-     * GET /files/attachments/access-token/{filename:.+} : Generates an access token that is valid for 30 seconds and given filename
+     * GET files/file-upload-exercises/:exerciseId/submissions/:submissionId/:filename/access-token : Generates an access token that is valid for 30 seconds and given exerciseId and submissionId
      *
-     * @param filename name of the file, the access token is for
+     * @param exerciseId ID of the exercise the submission belongs to
+     * @param submissionId ID of the submission the access token is for
+     * @param filename The filename of the file
      * @return The generated access token
      */
-    @GetMapping("files/attachments/access-token/{filename:.+}")
+    @GetMapping("files/file-upload-exercises/{exerciseId}/submissions/{submissionId}/{filename:.+}/access-token")
     @PreAuthorize("hasRole('USER')")
-    public ResponseEntity<String> getTemporaryFileAccessToken(@PathVariable String filename) {
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        // TODO: we should check that the filename actually exists in the context in Artemis, ideally we also verify that the user has access to this file already here
-        if (filename == null) {
+    public ResponseEntity<String> getTemporaryFileUploadExerciseSubmissionAccessToken(@PathVariable Long exerciseId, @PathVariable Long submissionId,
+            @PathVariable String filename) {
+        log.debug("REST request to get an access_token for a file upload exercise submission with exerciseId: {} and submissionId {}", exerciseId, submissionId);
+
+        Optional<FileUploadSubmission> optionalSubmission = fileUploadSubmissionRepository.findById(submissionId);
+        Optional<FileUploadExercise> optionalFileUploadExercise = fileUploadExerciseRepository.findById(exerciseId);
+        if (optionalSubmission.isEmpty() || optionalFileUploadExercise.isEmpty()) {
             return ResponseEntity.badRequest().build();
         }
-        try {
-            String temporaryAccessToken = tokenProvider.createFileTokenWithCustomDuration(authentication, 30, filename);
-            return ResponseEntity.ok(temporaryAccessToken);
+
+        FileUploadSubmission submission = optionalSubmission.get();
+        // user that submitted the exercise
+        Optional<User> userOfTheSubmission = ((StudentParticipation) submission.getParticipation()).getStudent();
+        if (userOfTheSubmission.isEmpty()) {
+            return ResponseEntity.badRequest().build();
         }
-        catch (IllegalAccessException e) {
+
+        User requestingUser = userRepository.getUser();
```

Comment:
Unsure about that one, but I remember having some problems with lazy loading here. So it might be better to explicity load all the required information.
```suggestion
        User requestingUser = userRepository.getUserWithGroupsAndAuthorities();
```

Reply:
adapted